x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/parser/LogicalPlanBuilder.java

```diff
@@ -521,8 +524,19 @@ private Expression onlyOnePipeArgument(Source source, String pipeName, List<Bool
 
     private Expression pipeIntArgument(Source source, String pipeName, List<BooleanExpressionContext> exps) {
         Expression expression = onlyOnePipeArgument(source, pipeName, exps);
+        boolean foldableInt = expression.dataType().isInteger() && expression.foldable();
+        boolean foldingError = false;
+        Number value = null;
+
+        if (foldableInt) {
+            try {
+                value = (Number) expression.fold();
+            } catch (ArithmeticException ae) {
+                foldingError = true;
+            }
+        }
 
-        if (expression.dataType().isInteger() == false || expression.foldable() == false || (int) expression.fold() < 0) {
+        if (foldableInt == false || foldingError || value.intValue() != value.longValue() || value.intValue() < 0) {
```

Comment:
Optional:
```
} else if (<value checks, knowing `foldableInt` is `false`>) {
```

Reply:
I'll keep it as is, I find it less cluttered.